src/Adapter/MSTest.TestAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -165,6 +180,15 @@ internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElem
 
                 discoverySink.SendTestCase(testCase);
             }
+
+            // If there are runnable tests, then add all fixture tests to the discovery sink.
+            if (hasAnyRunnableTests)
+            {
+                foreach (TestCase testCase in fixtureTests)
+                {
+                    discoverySink.SendTestCase(testCase);
```

Comment:
I think we are sending the entries twice (here and in the loop above) because you add to the list but don't skip the rest of the logic.

Reply:
Added code comments. In the above loop we put fixture methods in a separate list and do not add. They are added here.